images/manipulation.go

```diff
@@ -84,3 +108,107 @@ func CropCenter(img image.Image) (image.Image, error) {
 	}
 	return Crop(img, cropRect)
 }
+
+func ImageToBytes(imagePath string) ([]byte, error) {
+	// Open the image file
+	file, err := os.Open(imagePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	// Decode the image
+	img, _, err := image.Decode(file)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new buffer to hold the image data
+	var imgBuffer bytes.Buffer
+
+	// Encode the image to the desired format and save it in the buffer
+	err = png.Encode(&imgBuffer, img)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return the image data as a byte slice
+	return imgBuffer.Bytes(), nil
+}
+
+func AddPadding(img image.Image, padding int) *image.RGBA {
+	bounds := img.Bounds()
+	newBounds := image.Rect(bounds.Min.X-padding, bounds.Min.Y-padding, bounds.Max.X+padding, bounds.Max.Y+padding)
+	paddedImg := image.NewRGBA(newBounds)
+	draw.Draw(paddedImg, newBounds, &image.Uniform{C: color.White}, image.ZP, draw.Src)
+
+	return paddedImg
+}
+
+func EncodePNG(img *image.RGBA) ([]byte, error) {
+	resultImg := &bytes.Buffer{}
+	err := png.Encode(resultImg, img)
+	if err != nil {
+		return nil, err
+	}
+	return resultImg.Bytes(), nil
+}
+
+func CreateCircle(img image.Image) *image.RGBA {
+	bounds := img.Bounds()
+	circle := image.NewRGBA(bounds)
+	draw.DrawMask(circle, bounds, img, image.ZP, &Circle{
+		X: bounds.Dx() / 2,
+		Y: bounds.Dy() / 2,
+		R: bounds.Dx() / 2,
+	}, image.ZP, draw.Over)
+	return circle
+}
+
+func PlaceCircleInCenter(paddedImg, circle *image.RGBA) *image.RGBA {
+	bounds := circle.Bounds()
+	centerX := (paddedImg.Bounds().Min.X + paddedImg.Bounds().Max.X) / 2
+	centerY := (paddedImg.Bounds().Min.Y + paddedImg.Bounds().Max.Y) / 2
+	draw.Draw(paddedImg, bounds.Add(image.Pt(centerX-bounds.Dx()/2, centerY-bounds.Dy()/2)), circle, image.ZP, draw.Over)
+	return paddedImg
+}
+
+func ResizeImage(imgBytes []byte, width, height int) ([]byte, error) {
+	// Decode image bytes
+	img, _, err := image.Decode(bytes.NewReader(imgBytes))
+	if err != nil {
+		return nil, err
+	}
+	// Create a new image with the desired dimensions
+	newImg := image.NewNRGBA(image.Rect(0, 0, width, height))
+	xdraw.BiLinear.Scale(newImg, newImg.Bounds(), img, img.Bounds(), draw.Over, nil)
+	// Encode the new image to bytes
+	var newImgBytes bytes.Buffer
+	if err = png.Encode(&newImgBytes, newImg); err != nil {
+		return nil, err
+	}
+	return newImgBytes.Bytes(), nil
+}
+
+func SuperimposeLogoOnQRImage(imageBytes []byte, qrFilepath []byte) []byte {
+	// Read the two images from bytes
+	img1, _, _ := image.Decode(bytes.NewReader(imageBytes))
+	img2, _, _ := image.Decode(bytes.NewReader(qrFilepath))
```

Comment:
If I'm not mistaken `image.Decode()` returns an error as its 3rd return value. These errors should be handled

Reply:
will do 🫡

Reply:
fixed here : https://github.com/status-im/status-go/pull/3172/commits/35fd121cfa2729b0ab9c85f9e482408047609fd6